components/registry-facade/pkg/registry/metrics.go

```diff
@@ -0,0 +1,71 @@
+package registry
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// NewMeasuringRegistryRoundTripper produces a round tripper that exposes registry access metrics
+func NewMeasuringRegistryRoundTripper(delegate http.RoundTripper, reg prometheus.Registerer) (http.RoundTripper, error) {
+	metrics, err := newMetrics(reg)
+	if err != nil {
+		return nil, err
+	}
+	return &measuringRegistryRoundTripper{
+		delegate: delegate,
+		metrics:  metrics,
+	}, nil
+}
+
+type measuringRegistryRoundTripper struct {
+	delegate http.RoundTripper
+	metrics  *metrics
+}
+
+func (m *measuringRegistryRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	t0 := time.Now()
+	resp, err := m.delegate.RoundTrip(req)
+	dt := time.Since(t0)
+
+	if strings.Contains(req.URL.Path, "/manifests/") {
+		m.metrics.ManifestHist.Observe(dt.Seconds())
+	} else if strings.Contains(req.URL.Path, "/blobs/") {
+		m.metrics.BlobCounter.Inc()
+	}
+
+	return resp, err
+}
+
+// Metrics combine custom metrics exported by registry facade
+type metrics struct {
+	ManifestHist prometheus.Histogram
+	BlobCounter  prometheus.Counter
+}
+
+func newMetrics(reg prometheus.Registerer) (*metrics, error) {
+	manifestHist := prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name:    "manifest_req_sec",
```

Comment:
Is there any strong reason why this metric ends with `_sec` instead of `_seconds`?